### PR TITLE
Default event visibility

### PR DESCRIPTION
**Pull request by rufuswilson**
Some of my events had a visibility class of private and I wanted to be able to override it to the calendar setting.

### CODE REVIEW

File: Code.gs [hunk 1]
```diff
@@ -41,6 +41,7 @@ var addCalToTitle = false;             // Whether to add the source calendar to
 var addAttendees = false;              // Whether to add the attendee list. If true, duplicate events will be automatically added to the attendees' calendar.
 var defaultAllDayReminder = -1;        // Default reminder for all day events in minutes before the day of the event (-1 = no reminder, the value has to be between 0 and 40320)
                                        // See https://github.com/derekantrican/GAS-ICS-Sync/issues/75 for why this is neccessary.
+var forceVisibility = false;           // Changes the visibility of the event ("default", "public", "private"). Anything else will revert to the class value of the ICAL event.
```
Comment (derekantrican) [inline Code.gs:44]: Let's call this `override` rather than `force`

File: Helpers.gs [hunk 2]
```diff
@@ -355,10 +355,14 @@ function createEvent(event, calendarTz){
   if (event.hasProperty('location'))
     newEvent.location = icalEvent.location;
 
-  if (event.hasProperty('class')){
-    var classString = event.getFirstPropertyValue('class').toString().toLowerCase();
-    if (["default", "public", "private", "confidential"].indexOf(classString) > -1)
-      newEvent.visibility = classString;
+  if ( ["default", "public", "private"].includes(forceVisibility.toLowerCase()) ) {
```
Comment (derekantrican) [inline Helpers.gs:358]: Why does this array not include `confidential` like the array on 363? I think better to add that value and break out the array to something like `var validVisibilityValues` and use it in both places

File: Helpers.gs [hunk 3]
```diff
@@ -355,10 +355,14 @@ function createEvent(event, calendarTz){
...
+    newEvent.visibility = forceVisibility;
+  } else {
+    if (event.hasProperty('class')){
```
Comment (derekantrican) [inline Helpers.gs:361]: This `if` can be combined with the `else` above (ie `else if`). No need to have them separate.

**Review by derekantrican — CHANGES_REQUESTED** [review]

File: Code.gs (hunk 1 above)
Comment (rufuswilson) [reply Code.gs]: Agreed. Changes committed.

File: Helpers.gs (hunk 2 above)
Comment (rufuswilson) [reply Helpers.gs]: Agreed. Changes committed.

File: Helpers.gs (hunk 3 above)
Comment (rufuswilson) [reply Helpers.gs]: agreed and done

File: Code.gs [hunk 4]
```diff
@@ -41,6 +41,7 @@ var addCalToTitle = false;             // Whether to add the source calendar to
 var addAttendees = false;              // Whether to add the attendee list. If true, duplicate events will be automatically added to the attendees' calendar.
 var defaultAllDayReminder = -1;        // Default reminder for all day events in minutes before the day of the event (-1 = no reminder, the value has to be between 0 and 40320)
                                        // See https://github.com/derekantrican/GAS-ICS-Sync/issues/75 for why this is neccessary.
+var overrideVisibility = false;           // Changes the visibility of the event ("default", "public", "private"). Anything else will revert to the class value of the ICAL event.
```
Comment (derekantrican) [inline Code.gs:44]: Thanks for updating this. Even though it doesn't matter much from the perspective of someone familiar with Javascript, let's change the default value from `false` to a string (like `""` you had before) to make it more obvious to users. I feel like if it should be set to a string value, the default value should probably also be a string

Comment (rufuswilson) [reply Code.gs]: OK

**Review by derekantrican — APPROVED** [review]

**Pull request merged**